http/1.6/modules/ROOT/pages/http-request-ref.adoc

```diff
@@ -175,15 +175,47 @@ A range of failure status codes is defined by two ASCII `..` full stop character
 
-== Configure Request Streaming
+== Configure Request Streaming Mode
 
-By default, if the type of the payload is a stream, streaming is used to send the request. You can change this default behavior. Select *General > Request > Request Streaming* with one of the following values:
+When HTTP Connector manages request bodies, the connector considers the type of data to send. You can configure the HTTP *Request* operation to send HTTP requests in chunks. By default, if the payload type is stream, the operation uses the streaming mechanism to send HTTP requests. +
+When streaming, the HTTP request does not contain the `Content-Length` header. The request contains the `Transfer-Encoding` header and sends the body in chunks until the stream is fully consumed.
 
-* *AUTO* (default): The behavior depends on the payload type. If the payload is an InputStream, then streaming is enabled, otherwise it is disabled.
-* *ALWAYS*: Always enable streaming regardless of the payload type.
-* *NEVER*: Never stream, even if the payload is a stream.
+To change the default behavior, configure *Request streaming mode* to any of these options:
 
-When streaming, the request does not contain the `Content-Length` header. It contains the `Transfer-Encoding` header and sends the body in chunks until the stream is fully consumed.
+* *ALWAYS* +
+Always enable streaming regardless of the payload type.
+* *AUTO* (default) +
+The behavior depends on the payload type. If the payload is stream, streaming is enabled, otherwise it is disabled.
+* *NEVER* +
+Never stream, even if the payload is a stream.
+
+The following example shows how to configure  *Request streaming mode* in Studio:
+
+. In the *Mule Palette* view, select *HTTP > Request*.
+. Drag *Request* to the Studio canvas.
+. Set *Path* to `/path`.
+. In the HTTP *Request* operation configuration screen, click the *Request* tab.
+. Set *Request streaming mode* to `ALWAYS`.
+
+image::http-request-6.png[HTTP Request configuration window with Response streaming mode field set to ALWAYS]
+
+In the Configuration XML editor, the `requestStreamingMode` configuration looks like this:
+
+[source,xml,linenums]
+----
+<flow name="httprequestFlow"  >
+		<http:request method="GET" requestStreamingMode="ALWAYS" config-ref="HTTP_Request_configuration" path="/path"/>
+	</flow>
+----
+
+=== Configure Request Streaming System Property
+
+The previous *Request streaming mode* configuration causes the Mule application to send HTTP requests in chunks, but the configuration doesn't imply that the Mule app streams the payload internally. The HTTP *Request* operation needs all the data in memory before sending HTTP requests. Therefore, if the Mule app needs to send a huge request body, the Mule app can generate a high memory consumption, and eventually cause an out-of-memory error. To prevent such an issue, Mule provides the `mule.http.requestStreaming.enable` system property to enable internal request streaming.
```

Comment:
Inconsistent term - app or application

Reply:
```suggestion
The previous *Request streaming mode* configuration causes the Mule application to send HTTP requests in chunks, but the configuration doesn't imply that the Mule app streams the payload internally. The HTTP *Request* operation needs all the data in memory before sending HTTP requests. Therefore, if the Mule app needs to send a huge request body, the Mule app can generate a high memory consumption, and eventually cause an out-of-memory error. To prevent this issue, Mule provides the `mule.http.requestStreaming.enable` system property to enable internal request streaming.
```

Reply:
I suggest changing huge to very large.